src/lib/evil/unposix/stdio.c

```diff
@@ -0,0 +1,27 @@
+#include "stdio.h"
+#include <stdlib.h>
+#include <stdarg.h>
+
+#define vscprintf _vscprintf
+#define vsnprintf _vsnprintf
+
+
+int
+vasprintf(char **strp, const char *fmt, va_list ap)
+{
+   int ret = -1;
+
+   int len = vscprintf(fmt, ap);
+   if (len != -1)
+     {
+        char *str = (char *)malloc((len + 1) * sizeof(char));
+        if (str != NULL) 
+          {
+             ret = vsnprintf(str, (unsigned long) len + 1, fmt, ap);
+             if (ret == len) *strp = str;
+             else            free(strp);
```

Comment:
why so much space?

Reply:
I personally think it makes easier to read, or at least forces me to read the else. Maybe it is just me... Should it be removed?

Reply:
Answering based on your, JPTIZ and Coquinho suggestions, I would avoid 'table statements' style.